interfaces.go

```diff
@@ -46,11 +44,12 @@ type DirectedGraph[NodeType any] interface {
 	Clone() DirectedGraph[NodeType]
 	// HasCycles performs cycle detection and returns true if the DirectedGraph has cycles.
 	HasCycles() bool
-	// PopReadyNodes lists all nodes that have finalized their status, whether resolved
-	// or unresolvable, and clears the list.
+	// PopReadyNodes returns of a list of all nodes that have finalized their status, whether
+	// resolved or unresolvable, and clears the list.
 	PopReadyNodes() []*node[NodeType]
-	// PushStartingNodes pushes the initial nodes without dependencies for access with PopReadyNodes()
-	// O(N), so use only at the start.
+	// PushStartingNodes searches for the initial ready nodes without dependencies and saves them.
+	// The nodes can then be retrieved with a call to `PopReadyNodes()`.
+	// Recommended to be called only once following construction of the DAG.
 	PushStartingNodes() error
```

Comment:
`InitStartingNodes()` would certainly better convey the intended one-time use of the function.

The notion that someone would call this function mid-processing because they can't get access to the result of a previous call to `pop` doesn't really seem like a use case we would want to make much effort to support (especially given our stated recommendation that it should only be called once...).

From the perspective of the use of the interface (i.e., `DirectedGraph`), the caller constructs the graph object, adds nodes, connects them, and then...calls "start"...then then basically loops, fetching the ready nodes, resolving them, and repeating until there are no more nodes (etc.).  So, "push" is really only interesting in opposition to "pop" (and "pop" is really only the choice because it's destructive, where using "get" would suggest non-destructive); and "nodes" isn't important from the caller's perspective (we're talking about transitioning the _DAG_ from the construction phase to the operational phase).  So, ideas like "init", "start", et al. are good; but I don't see how the other bits help (the calling side of the interface doesn't need to consider that the implementation of the function is doing stuff with nodes).